docs/build/building-apps/upgrades/0.52.md

```diff
@@ -0,0 +1,539 @@
+
+With the introduction of environment in modules, depinject automatically creates the environment for all modules. Learn more about environment here . Given the fields of environment, this means runtime creates a kv store service for all modules by default. It can happen that some modules do not have a store necessary (such as x/auth/tx for instance). In this case, the store creation should be skipped in app_config.go:
+
+* In your app_config.go (or app.yml), if you have modules like x/auth/tx that do not need a KV store, skip them:
+
+```diff
+InitGenesis: []string{
+	"..."
+},
++ // SkipStoreKeys is an optional list of store keys to skip when constructing the
++ // module's keeper. This is useful when a module does not have a store key.
++ SkipStoreKeys: []string{
++ 	"tx",
++ },
+```
+
+
+
+### Protobuf Changes
+
+The cosmossdk.io/api/tendermint package has been removed as CometBFT now publishes its protos to buf.build/tendermint and buf.build/cometbft
+
+
+The codectypes.Any has moved to github.com/cosmos/gogoproto/types/any. Module developers need to update the buf.gen.gogo.yaml configuration files by adjusting the corresponding opt option to Mgoogle/protobuf/any.proto=github.com/cosmos/gogoproto/types/any for directly mapping theAny type to its new location:
+
+```diff
+version: v1
+plugins:
+  - name: gocosmos
+    out: ..
+- 	 opt: plugins=grpc,Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types,Mcosmos/orm/v1/orm.proto=cosmossdk.io/orm
++    opt: plugins=grpc,Mgoogle/protobuf/any.proto=github.com/cosmos/gogoproto/types/any,Mcosmos/orm/v1/orm.proto=cosmossdk.io/orm
+  - name: grpc-gateway
+    out: ..
+    opt: logtostderr=true,allow_colon_final_segments=true
+
+```
+
+* Replace any usage of AnyUnpacker and UnpackInterfacesMessage from cosmos/cosmos-sdk/codec/types with the equivalents in github.com/cosmos/gogoproto/types/any.
+
+### Refactor Module Imports to cosmossdk.io/x/
+
+All modules except auth have been split into their own go.mod and are imported via cosmossdk.io/x/<mod>.
+
+
+* Replace import paths from github.com/cosmos/cosmos-sdk/x/{moduleName} to cosmossdk.io/x/{moduleName}.
+* Example:
+
+```diff
+- import govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
++ import govtypes "cosmossdk.io/x/gov/types"
+```
+
+### Adopt Core API Interfaces
+
+Core API has been introduced for modules since v0.47. With the deprecation of sdk.Context, we strongly recommend to use the cosmossdk.io/core/appmodule interfaces for the modules. This will allow the modules to work out of the box with server/v2 and baseapp, as well as limit their dependencies on the SDK.
+
+Additionally, the appmodule.Environment struct is introduced to fetch different services from the application. This should be used as an alternative to using sdk.UnwrapContext(ctx) to fetch the services. It needs to be passed into a module at instantiation (or depinject will inject the correct environment).
```

Comment:
```suggestion
Core API has been introduced for modules since v0.47. With the deprecation of `sdk.Context`, we strongly recommend to use the cosmossdk.io/core/appmodule interfaces for the modules. This will allow the modules to work out of the box with server/v2 and baseapp, as well as limit their dependencies on the SDK.

Additionally, the `appmodule.Environment` struct is introduced to fetch different services from the application. This should be used as an alternative to using `sdk.UnwrapContext(ctx)` to fetch the services. It needs to be passed into a module at instantiation (or depinject will inject the correct environment).
```